File: Samples/Basic/DocumentReader-sample/app/src/main/java/com/regula/documentreader/util/CertificatesUtil.java
package com.regula.documentreader.util;

import android.content.res.AssetManager;

import com.regula.documentreader.api.enums.PKDResourceType;
import com.regula.documentreader.api.params.rfid.PKDCertificate;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class CertificatesUtil {

   public static List<PKDCertificate> getRfidCertificates(AssetManager am, String certificatesDir) {
      List<PKDCertificate> pkdCertificatesList = new ArrayList<>();
      try {
         String list[] = am.list(certificatesDir);
         if (list != null && list.length > 0) {
            for (String file : list) {
               String[] findExtension = file.split("\\.");
               int pkdResourceType = 0;
               if (findExtension.length > 0) {
                  pkdResourceType = PKDResourceType.getType(findExtension[findExtension.length - 1].toLowerCase());
               }

               InputStream licInput = am.open(certificatesDir+"/"+file);
               int available = licInput.available();
               byte[] binaryData = new byte[available];
               licInput.read(binaryData);

               PKDCertificate certificate = new PKDCertificate(binaryData, pkdResourceType);
               pkdCertificatesList.add(certificate);
            }
         }
      } catch (IOException e) {
         e.printStackTrace();
      }
      return pkdCertificatesList;
   }

   public static List<PKDCertificate> getRfidTACertificates(AssetManager am) {
      List<PKDCertificate> pkdCertificatesList = new ArrayList<>();
      String certificatesDir = "Regula/certificates_ta";
      try {
         Map<String, List<String>> filesCertMap = new HashMap<>();
         String list[] = am.list(certificatesDir);
         if (list != null && list.length > 0) {
            for (String file : list) {
               String[] findExtension = file.split("\\.");
               if (!filesCertMap.containsKey(findExtension[0])) {
                  List<String> certList = new ArrayList<>();
                  certList.add(file);
                  filesCertMap.put(findExtension[0], certList);
               } else {
                  filesCertMap.get(findExtension[0]).add(file);
               }
            }
         }

         for (Map.Entry me : filesCertMap.entrySet()) {
            List<String> files = (List<String>) me.getValue();
            PKDCertificate certificate = new PKDCertificate();
            for(String file : files) {
               String[] findExtension = file.split("\\.");
               certificate.resourceType = PKDResourceType.CERTIFICATE_TA;
               InputStream licInput = am.open(certificatesDir+"/"+file);
               int available = licInput.available();
               byte[] binaryData = new byte[available];
               licInput.read(binaryData);
               if (findExtension[1].equals("cvCert")) {
                  certificate.binaryData = binaryData;
               } else {
                  certificate.privateKey = binaryData;
               }
            }
            pkdCertificatesList.add(certificate);
         }

      } catch (IOException e) {
         e.printStackTrace();
      }
      return pkdCertificatesList;
   }
}

File: Samples/Basic/DocumentReader-sample_kotlin/app/src/main/java/com/regula/documentreader/util/CertificatesUtil.kt
package com.regula.documentreader.util

import android.content.res.AssetManager
import com.regula.documentreader.api.enums.PKDResourceType
import com.regula.documentreader.api.params.rfid.PKDCertificate
import java.io.IOException
import java.io.InputStream
import java.util.*

object CertificatesUtil {

    fun getRfidCertificates(am: AssetManager, certificatesDir: String): List<PKDCertificate> {
        val pkdCertificatesList: MutableList<PKDCertificate> = ArrayList()
        try {
            val list = am.list(certificatesDir)
            if (list != null && list.isNotEmpty()) {
                for (file in list) {
                    val findExtension = file.split("\\.").toTypedArray()
                    var pkdResourceType = 0
                    if (findExtension.isNotEmpty()) {
                        pkdResourceType =
                            PKDResourceType.getType(findExtension[findExtension.size - 1].toLowerCase())
                    }
                    val licInput = am.open("$certificatesDir/$file")
                    val available = licInput.available()
                    val binaryData = ByteArray(available)
                    licInput.read(binaryData)
                    val certificate = PKDCertificate(binaryData, pkdResourceType)
                    pkdCertificatesList.add(certificate)
                }
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return pkdCertificatesList

    }

    fun getRfidTACertificates(am: AssetManager): List<PKDCertificate>? {
        val pkdCertificatesList: MutableList<PKDCertificate> = ArrayList()
        val certificatesDir = "Regula/certificates_ta"
        try {
            val filesCertMap: MutableMap<String, MutableList<String>> = HashMap()
            val list = am.list(certificatesDir)
            if (list != null && list.isNotEmpty()) {
                for (file in list) {
                    val findExtension = file.split("\\.").toTypedArray()
                    if (!filesCertMap.containsKey(findExtension[0])) {
                        val certList: MutableList<String> = ArrayList()
                        certList.add(file)
                        filesCertMap[findExtension[0]] = certList
                    } else {
                        filesCertMap[findExtension[0]]!!.add(file)
                    }
                }
            }
            for ((_, value) in filesCertMap) {
                val files = value as List<String>
                val certificate = PKDCertificate()
                for (file in files) {
                    val findExtension = file.split("\\.").toTypedArray()
                    certificate.resourceType = PKDResourceType.CERTIFICATE_TA
                    val licInput: InputStream = am.open("$certificatesDir/$file")
                    val available: Int = licInput.available()
                    val binaryData = ByteArray(available)
                    licInput.read(binaryData)
                    if (findExtension[1] == "cvCert") {
                        certificate.binaryData = binaryData
                    } else {
                        certificate.privateKey = binaryData
                    }
                }
                pkdCertificatesList.add(certificate)
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return pkdCertificatesList
    }
}